feat: ratelimit prediction (#2188)
Pycord-Development/pycord: CHANGELOG.md

  ([#2036](https://github.com/Pycord-Development/pycord/pull/2036))
- Removed `Embed.Empty` in favour of `None`, and `EmbedProxy` in favour of individual
  classes. ([#2063](https://github.com/Pycord-Development/pycord/pull/2063))

> - Removed semi-private `MaybeUnlock` class as it is not in use anymore.

Reviewer: ```suggestion
- Removed the semi-private `MaybeUnlock` class as it is not in use anymore.
```

  ([#2099](https://github.com/Pycord-Development/pycord/pull/2099))
- Changed the support from `orjson` to `msgspec` in the codebase.
  ([#2170](https://github.com/Pycord-Development/pycord/pull/2170))

> - Replaced former rate limit system with a rate limit prediction and easing system.

Reviewer 2: ```suggestion
- Replaced former rate limit system with an easing prediction system.
```
Weird phrasing
